hack/ci/Dockerfile.ci

```diff
@@ -0,0 +1,130 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
+
+# Install packages
+# dnf copr enable -y @virtmaint-sig/virt-preview with yum?
+# python3-pip-9.0.3-5.el7.noarch \
+RUN yum install -y yum-plugin-copr && \
+    yum copr enable -y vbatts/bazel epel-7 && \
+    yum -y install \
+        libvirt-devel \
+        bazel \
+        cpio \
+        patch \
+        make \
+        git \
+        mercurial \
+        sudo \
+        gcc \
+        gcc-c++ \
+        glibc-devel \
+        findutils \
+        qemu-img \
+        protobuf-compiler \
+        python3-devel \
+        python3-pip \
+        redhat-rpm-config \
+        jq \
+        unzip && \
+    yum clean all && rm -rf /var/cache/yum/*
+
+ENV GIMME_GO_VERSION=1.12.8
+
+RUN mkdir -p /gimme && curl -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | HOME=/gimme bash >> /etc/profile.d/gimme.sh
+
+ENV GOPATH="/go" GOBIN="/usr/bin"
+
+# Install persistent go packages
+RUN \
+    mkdir -p /go && \
+    source /etc/profile.d/gimme.sh && \
+    # Install goimports
+    go get -u -d golang.org/x/tools/cmd/goimports && \
+    cd $GOPATH/src/golang.org/x/tools/cmd/goimports && \
+    git checkout release-branch.go1.12 && \
+    go install && \
+    # Install mvdan/sh
+    git clone https://github.com/mvdan/sh.git $GOPATH/src/mvdan.cc/sh && \
+    cd $GOPATH/src/mvdan.cc/sh/cmd/shfmt && \
+    git checkout v2.5.0 && \
+    go get mvdan.cc/sh/cmd/shfmt && \
+    go install && \
+    go get -u -d k8s.io/code-generator/cmd/deepcopy-gen && \
+    go get -u -d k8s.io/code-generator/cmd/defaulter-gen && \
+    go get -u -d k8s.io/kube-openapi/cmd/openapi-gen && \
+    go get -u -d github.com/golang/protobuf/protoc-gen-go && \
+    # Install deepcopy-gen
+    cd $GOPATH/src/k8s.io/code-generator/cmd/deepcopy-gen && \
+    git checkout kubernetes-1.13.4 && \
+    go install && \
+    # Install defaulter-gen
+    cd $GOPATH/src/k8s.io/code-generator/cmd/defaulter-gen && \
+    git checkout kubernetes-1.13.4 && \
+    go install && \
+    # Install openapi-gen
+    cd $GOPATH/src/k8s.io/kube-openapi/cmd/openapi-gen && \
+    git checkout c59034cc13d587f5ef4e85ca0ade0c1866ae8e1d && \
```

Comment:
What about doing all this at a bash script and use COPY directive ? it's going to make a layer too, well maybe two one for the script and one for the run of the script.

Reply:
What is the advantage of doing this in a bash script over doing it here? I would find it easier to do the cleanup later on when comparing it to the [builder file](https://github.com/kubevirt/kubevirt/blob/master/hack/builder/Dockerfile) if we'd leave it like this.